File: Clasde01/app/src/main/java/com/jeancarlo/myapplication/MainActivity.java
package com.jeancarlo.myapplication;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    Button btn;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        Log.d("Ejercicio", "onCreate");

        btn = findViewById(R.id.btnMover);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, Destino.class);
                startActivity(intent);
            }
        });
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d("Ejercicio", "onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d("Ejercicio", "onPause");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d("Ejercicio", "onStart");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d("Ejercicio", "onStop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d("Ejercicio", "onDestroy");
    }
}

File: Clasde01kt/app/src/main/java/com/jeancarlo/myapplication/MainActivity.kt
package com.jeancarlo.myapplication

import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.jeancarlo.myapplication.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var  binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        //setContentView(R.layout.activity_main)
        enableEdgeToEdge()
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        Log.d("ejercicioKT","onCreate")

        /* evento click kt */
        binding.btnMove.setOnClickListener {
            val intent = Intent(this, Destino::class.java)
            startActivity(intent)
        }

        binding.txtHola.text = "Soy nuevo en KT"
    }

    override fun onResume() {
        super.onResume()
        Log.d("ejercicioKT","onResume")
    }

    override fun onPause() {
        super.onPause()
        Log.d("ejercicioKT","onPause")
    }

    override fun onStart() {
        super.onStart()
        Log.d("ejercicioKT","onStart")
    }

    override fun onStop() {
        super.onStop()
        Log.d("ejercicioKT","onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("ejercicioKT","onDestroy")
    }
}